bliss/models/encoder.py

```diff
@@ -1,4 +1,5 @@
 import numpy as np
+from einops import rearrange
```

Comment:
einops looks useful, but there's also some cost to introducing new dependencies -- especially if only some of our code is using it. @ismael-mendoza Would you take a look at [einops](http://einops.rocks/pytorch-examples.html) too? Do we want to use it throughout bliss?

Reply:
Other than the one in this PR and the Multihead attention as shown [here](http://einops.rocks/pytorch-examples.html),
I just checked we have around 60+ of `torch.reshape`/`torch.view` in our code base. 
For example https://github.com/prob-ml/bliss/blob/d6cb336691de0d2a4d337dbe2b8c0a9cee8f27f2/bliss/metrics.py#L110
can be changed to something like 

```python
est_n_gal = einops.reduce(true_params["galaxy_bool"], "b n t -> b", "sum")
```
so that it's very clear which axes are being summed over

If `einops`'s syntax looks appealing , I can make another PR and change them altogether. But I agree, this is another layer of dependency that we may not want to add.

Reply:
That's very nice syntax. The tiling code we have is really complex now, and it seems like einop could make it much more readable. If you're up for submitting a second pr that update the rest of our codebase to eigop I'm very supportive of this change.

Reply:
Thanks @ZachariahPang for looking into this, it seems that `einops` is _really_ light weight (does it even have any dependencies other than python itself?). I'm also supportive since it's a lot more readable from what I can see.